imports/plugins/included/taxes-avalara/client/components/avalaraSettingsForm.js

```diff
@@ -0,0 +1,182 @@
+  * @property {Object} shownFields - info about the fields of the Avalara Package
+  * that the settings form will allow users to change.
+  * @property {Object} shownLogFields - fields to show from Log schema
+  * @return {Array} React propTypes
+  */
+  static propTypes = {
+    fieldsProp: PropTypes.object,
+    handleSubmit: PropTypes.func,
+    handleTestCredentials: PropTypes.func,
+    hiddenFields: PropTypes.arrayOf(PropTypes.string),
+    logFieldsProp: PropTypes.object,
+    settings: PropTypes.object,
+    shownFields: PropTypes.object,
+    shownLogFields: PropTypes.object
+  };
+  static filteredFields = [ "data.request.data.date", "data.request.data.type" ];
+  static noDataMessage = i18next.t("logGrid.noLogsFound");
+
+  // helper adds a class to every grid row
+  static customRowMetaData = {
+    bodyCssClassName: () =>  {
+      return "log-grid-row";
+    }
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      editingId: undefined,
+      showLogs: this.props.settings.avalara.enableLogging,
+      log: undefined,
+      fieldsProp: assign(clone(props.fieldsProp), {
+        "settings.avalara.enableLogging": {
+          handleChange: this.handleLogToggle
+        }
+      })
+    };
+  }
+
+  handleLogToggle = () => {
+    this.setState(() => ({ showLogs: !this.state.showLogs }));
+  }
+
+  handleTestCredentials = (event) => {
+    this.props.handleTestCredentials(event, this.props.settings);
+  }
+
+  editRow = (options) => {
+    const currentId = this.state.editingId;
+    this.setState(() => {
+      const id = options.props.data._id;
+      const log = Logs.findOne(id) || {};
+      log.data = JSON.stringify(log.data, null, 4);
+      return {
+        editingId: id,
+        log
+      };
+    });
+    // toggle edit mode clicking on same row
+    if (currentId === options.props.data._id) {
+      this.setState({
+        editingId: undefined,
+        log: undefined
+      });
+    }
+  }
+
+  render() {
+    const { handleSubmit, hiddenFields, settings, shownFields, shownLogFields } = this.props;
+
+    // add i18n handling to headers
+    const customColumnMetadata = AvalaraSettingsForm.filteredFields.reduce((arr, field) => {
+      arr.push({
+        accessor: field,
+        Header: i18next.t(`logGrid.columns.${field}`)
+      });
+      return arr;
+    }, []);
+
+
+    return (
+      <div className="rui avalara-update-form">
+        {!settings.avalara.apiLoginId &&
+          <div className="alert alert-info">
+            <Components.Translation defaultValue="Add API Login ID to enable" i18nKey="admin.taxSettings.avalaraCredentials" />
+            <a href="https://admin-development.avalara.net" target="_blank"> Avalara</a>
+          </div>
+        }
+        <Form
+          schema={AvalaraPackageConfig}
+          doc={{ settings }}
+          renderFromFields={true}
+          fieldsProp={this.state.fieldsProp}
+          docPath={"settings.avalara"}
```

Comment:
Do not wrap string prop values in `{}`

Reply:
@aldeed Would doing so considered to be bad practice? Just asking, because Webstorm is inserting brackets automatically and it would be tedious to remove them every time I'm going to insert string literals. Not sure if this is configurable.